app/src/main/kotlin/io/intrepid/fablekotlin/screens/createfable/CreateFableActivity.kt

```diff
@@ -70,6 +105,117 @@ class CreateFableActivity : BaseMvpActivity<CreateFableContract.Presenter>(), Cr
         pickedColor.setColorFilter(Color.parseColor(clickedColor.hexColor))
     }
 
+    @OnClick(R.id.circle_button1, R.id.circle_button2, R.id.circle_button3, R.id.circle_button4, R.id.circle_button5, R.id.circle_button6)
+    fun onClickCircleButton() {
+        val intent = Intent(this, SelectFriendsActivity::class.java)
```

Comment:
The onclick should definitely pass through your presenter. We also usually will build out a static intent builder whenever we need to use Extras. So the SelectFriendsActivity would have a static function that takes in all the parameters you need. That way you can guarantee that you have the information you need.